forloop_modules/function_handlers/input_waiter.py

```diff
@@ -0,0 +1,29 @@
+# input_waiter.py
+from forloop_modules.globals.variable_handler import LocalVariable, variable_handler
+from forloop_modules.queries.node_context_requests_backend import post_console_print_log
+
+_pending: dict[str, str] = {}
+_prompts: dict[str, str] = {} 
+
+def register_pending_input(node_uid: str, output_variable: str, prompt_text: str = None):
+    _pending[node_uid] = output_variable
+    if prompt_text:
+        _prompts[node_uid] = prompt_text
+
+def resolve_input(node_uid: str, value: str):
+    var_name = _pending.pop(node_uid, None)
+    prompt_text = _prompts.pop(node_uid, None) 
+
+    if not var_name:
+        return
+
+    if var_name in variable_handler.variables:
+        variable_handler.variables[var_name] = LocalVariable(uid=0, name=var_name, value=value, is_result=False)
```

Comment:
LocalVariable object shouldnt be initiated outside variable_handler

Reply:
if you need new variable you should have probably used variable_handler.new_variable(......)

Reply:
I dont get why is here this whole if condition, it wouldnt work just with what is in else branch?